src/CocktailDetails/CocktailDetails.test.tsx

```diff
@@ -1,70 +1,181 @@
+          id={"14622"}
+          favCocktails={[]}
+          setFavCocktails={Function}
+          toggleUserInteraction={mockToggle}
+          madeCocktails={[]}
+          setMadeCocktails={Function}
+        />
+      </MemoryRouter>
+    );
 
-		const ingredient1 = await waitFor(() => getByText('Vodka 1/3 part'));
-		const ingredient2 = await waitFor(() => getByText('Grape soda 1/3 part'));
-		const ingredient3 = await waitFor(() => getByText('Orange juice 1/3 part'));
-		const ingredient4 = await waitFor(() => getByText('Ice lots'));
-		const ingredient5 = await waitFor(() => getByText('Candy 1 dash'));
+    const ingredient1 = await waitFor(() => getByText("Vodka"));
+    const strMeasure1 = await waitFor(() => getAllByText("1/3 part"));
+    const ingredient2 = await waitFor(() => getByText("Grape soda"));
+    const ingredient3 = await waitFor(() => getByText("Orange juice"));
+    const ingredient4 = await waitFor(() => getByText("Ice"));
+    const strMeasure4 = await waitFor(() => getByText("lots"));
+    const ingredient5 = await waitFor(() => getByText("Candy"));
+    const strMeasure5 = await waitFor(() => getByText("1 dash"));
 
-		expect(ingredient1).toBeInTheDocument();
-		expect(ingredient2).toBeInTheDocument();
-		expect(ingredient3).toBeInTheDocument();
+    expect(ingredient1).toBeInTheDocument();
+    expect(strMeasure1.length).toEqual(3);
+    expect(ingredient2).toBeInTheDocument();
+    expect(ingredient3).toBeInTheDocument();
 		expect(ingredient4).toBeInTheDocument();
+		expect(strMeasure4).toBeInTheDocument();
 		expect(ingredient5).toBeInTheDocument();
+		expect(strMeasure5).toBeInTheDocument();
 	});
-})
\ No newline at end of file
+	
+	it("Should be able to click add to favorites button", () => {
+		const mockToggle = jest.fn();
+		const mockFavorite = jest.fn();
+
+    const { getByLabelText, getByText } = render(
+      <MemoryRouter>
+        <CocktailDetails
+          id={"14622"}
+          favCocktails={[]}
+          setFavCocktails={mockFavorite}
+          toggleUserInteraction={mockToggle}
+          madeCocktails={[]}
+          setMadeCocktails={Function}
+        />
+      </MemoryRouter>
+		);
+
+		const favButton = getByText('Add to Favorites')
+
+		fireEvent.click(favButton)
+
+		expect(mockToggle).toHaveBeenCalled()
+		expect(getByText("Remove from Favorites")).toBeInTheDocument()
+	})
+
+	it("Should be able to click add to Make button", () => {
+		const mockToggle = jest.fn();
+		const mockFavorite = jest.fn();
+
+    const { getByLabelText, getByText } = render(
+      <MemoryRouter>
+        <CocktailDetails
+          id={"14622"}
+          favCocktails={[]}
+          setFavCocktails={mockFavorite}
+          toggleUserInteraction={mockToggle}
+          madeCocktails={[]}
+          setMadeCocktails={Function}
+        />
+      </MemoryRouter>
+		);
+
+		const favButton = getByText('Mark as Made')
+
+		fireEvent.click(favButton)
+
+		expect(mockToggle).toHaveBeenCalled()
```

Comment:
This was tricky to figure how to test if a button was clicked, as the actual setMadeCocktails was handled through a clickHandler, and had to figure out which function to test - ie mockToggle.

Reply:
Nice job figuring this out! That would have taken me forever.

Reply:
nice job!